src/plugin.js

```diff
@@ -0,0 +1,17 @@
+import * as components from './components';
+
+export default function (Vue, options) {
+  // auto-import all components
+  Vue.mixin({
+    components: { ...components }, // casting Module to Object
```

Comment:
Now that I think about it, I think auto registering all components for all Vue instances in the app is actually a bit weird. The usual usage would be more on something like `BaseButton` which would be used across the whole app, and it's discouraged to register components which are not eventually used. But yeah since we normally only have one `App.vue` anyways I think this might still make sense. Just wonder if this would have an impact on load time, but I guess even if that's the case it shouldn't be huge.